### ISSUE TITLE
SPINE and mysql

### ISSUE THREAD

**Issue by interduo** [Cacti/spine#36]
```
2017/11/26  20:30:00 - SYSTEM STATS: Time:298.9212 Method:spine Processes:1  Threads:20 Hosts:455 HostsPerProcess:455 DataSources:6619  RRDsProcessed:1329
--
2017/11/26 20:30:00 - POLLER: Poller[1] Maximum runtime of 298 seconds exceeded. Exiting.
2017-11-26 20:25:25 - SPINE: Poller[1] FATAL: MySQL Error:'2006',  Message:'MySQL server has gone away' [110, Connection timed out] (Spine  thread)
2017-11-26 20:25:25 - SPINE: Poller[1] ERROR: SQL Failed!  Error:'2013', Message:'Lost connection to MySQL server during query',  SQL Fragment:'INSERT INTO poller_output (local_data_id, rrd_name, time,  output) VALUES (2454,'ping','2017-11-26 20:25:13','13.7') ON DUPLICATE  KEY UPDATE output=VALUES(output)'
```

1. Time stamps "/" instead of "-".
2. Thats not true what SPINE says 
```
$ ps aux | grep mysqld
mysql    25283  0.0 24.8 2465200 1040744 ?     Ssl  paź14 1667:01 /usr/sbin/mysqld

```

**Comment by cigamit** [comment]
This server has gone away issue was related to an API issue in older version of MySQL.  What version of MySQL is this?  Is this MySQL on the same server as the poller?  How often does this happen?

**Comment by interduo** [comment]
MySQL is on the same server as poller.

> # mysql --version
> mysql  Ver 15.1 Distrib 10.1.26-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
>

**Comment by cigamit** [comment]
How often is this happening?

**Comment by interduo** [comment]
Not often so it's hard to debug. One/two time per week. I did a research and the resources and not fully used.

**Comment by cigamit** [comment]
Well, this is going to be very hard to debug then.  There was an issue in early MySQL's and correspondingly in MariaDB's protocol stack where the API was being interrupted due to a very low level protocol communications handling error.  In other words, the connection did not go away, but MySQL/MariaDB thought it did.  We finally got this issue fixed about 8 months or so ago.  It's in all the mainline MySQL 5.5/5.6/5.7 now.  Not sure about MariaDB versions and point releases where it was fixed though.
